src/GameLogic/DataStructures/EntityMovement.ts

```diff
@@ -0,0 +1,24 @@
+import {Entity} from '../Entity';
+
+export class EntityMovement {
+	public entity: Entity;
+
+	public oldX: number;
+
+	public oldY: number;
+
+	public newX: number | undefined;
```

Comment:
How about instead of making it undefinable, make it so that newX/Y is set to oldX/Y in the constructor when they are not provided? This will mean we won't have to worry about the special case of undefined in the code and technically the new position is the same as old

Reply:
The original plan here was that the entities also worked out if their intended move would kill them, then they would neither be at their old location or any new location. This would allow e.g. two protagonists to drown on the same water tile. But given the plan is they will die anyway I took this out, but didn't go back and change this.